components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/exception/SessionDataStorageOptimizationException.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.exception;
+
+/**
+ * This exception is used to handle the exceptions which occurred during optimization process or loading process of
+ * session data storage optimization.
+ */
+public class SessionDataStorageOptimizationException extends FrameworkException {
```

Comment:
we discussed to have a client exception as well to isolate the idp and application null cases and log as debug 
and for other server errors, log as error logs

Reply:
Resolved with [`646a5dd`](https://github.com/wso2/carbon-identity-framework/pull/4446/commits/646a5dd851a1d454c0d80813f913fc7f4beb71d7)